test/cql-pytest/test_secondary_index.py

```diff
@@ -64,3 +64,32 @@ def test_partition_order_with_si(cql, test_keyspace):
                 pass
             time.sleep(0.1)
         assert tokens2 == tokens
+
+# Test which ensures that indexes for a query are picked
+# by the order in which they appear in restrictions.
+# That way, users can deterministically pick which indexes
+# are used for which queries.
+# Ref: #7969
+@pytest.mark.xfail(reason="The order of picking indexes is currently arbitrary. Issue #7969")
```

Comment:
I'm curious how, if this test fails, did you check that the test itself is correct.
Also, the test has a lot of subtests - which of them are actually expected to fail? All of them? One of them? Some of them? If one specific check causes everything to fail, you can add a comment saying that this specific line reproduces #7969. But if all of them fail, maybe you can say this somewhere too.

Reply:
it was honest, arduous manual labor :) I commented out everything and enabled each case separately, verifying that the queries work and use indexes, but in some cases not the one we expect, assuming the order. I'll add a comment stating that *all* cases below should pass, while right now just some of them do (the order is kind of arbitrary and depends e.g. on whether Scylla was restarted after adding the indexes, so no guarantees which test cases fail and which don't).